optuna/integration/wandb.py

```diff
@@ -74,28 +74,45 @@ def objective(trial):
             <https://docs.wandb.ai/ref/python/init>`_ for more details.
 
     Raises:
-        :exc:`RuntimeError`:
+        :exc:`ValueError`:
             If there are missing or extra metric names in multi-objective optimization.
+        :exc:`TypeError`:
+            When metric names are not passed as sequence.
     """
 
     def __init__(
         self,
-        metric_name: Union[str, List[str]] = "value",
+        metric_name: Union[str, Sequence[str]] = "value",
         wandb_kwargs: Optional[Dict[str, Any]] = None,
     ) -> None:
 
         _imports.check()
 
+        if not isinstance(metric_name, Sequence):
+            raise TypeError(
+                "Expected metric_name to be string or sequence of strings, got {}.".format(
+                    type(metric_name)
+                )
+            )
+
         self._metric_name = metric_name
         self._wandb_kwargs = wandb_kwargs or {}
 
         self._initialize_run()
 
     def __call__(self, study: optuna.study.Study, trial: optuna.trial.FrozenTrial) -> None:
 
-        if isinstance(self._metric_name, list):
+        if isinstance(self._metric_name, str):
+            if len(trial.values) > 1:
+                # Broadcast default name for multi-objective optimization.
+                names = ["{}_{}".format(self._metric_name, i) for i in range(len(trial.values))]
+
+            else:
+                names = [self._metric_name]
+
+        else:
             if len(self._metric_name) != len(trial.values):
-                raise RuntimeError(
+                raise ValueError(
```

Comment:
Note: If we change the error type, users will usually need to update their code. In this PR, I think it is acceptable to change the type without messages for backward compatibility because the callback is marked as experimental and users do not usually catch this `RuntimeError`.